src/core/content/connector.ts

```diff
@@ -614,6 +643,29 @@ export default class BaseConnector {
 		this._controllerCallback = callback;
 	}
 
+	/**
+	 * Callback set by the controller to listen on state changes of this connector.
+	 */
+	private _isLovedCallback:
+		| ((isLoved: boolean | null) => Promise<void>)
+		| null = null;
+
+	/**
+	 * Callback set by the controller to listen on state changes of this connector.
+	 */
+	public get isLovedCallback():
+		| ((isLoved: boolean | null) => Promise<void>)
+		| null {
+		return this._isLovedCallback;
+	}
+
+	public set isLovedCallback(
+		callback: (isLoved: boolean | null) => Promise<void>,
+	) {
+		callback(this.isLoved() ?? null);
```

Comment:
is the `?? null` to ensure either `bool` or `null`?

Reply:
Correct, I allow connectors to return undefined on no result to be more lenient on connector creators, in line with other connector functions.